lib/src/scope.dart

```diff
@@ -2,81 +2,115 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-part of code_builder;
+import 'package:analyzer/analyzer.dart';
 
-/// Determines an [Identifier] deppending on where it appears.
+import 'builders/file.dart';
+import 'tokens.dart';
+
+/// Returns an identifier for [name], using [scope] to enforce prefixing.
+///
+/// It is _required_ within `code_builder` to use this API instead of creating
+/// identifeirs manually in places where an obvious API collision could occur
+/// (i.e. imported references).
+///
+/// If [scope] is null, no prefixing is applied.
+///
+/// ## Example
+///     // May output `hello.Hello`.
+///     identifier(scope, 'Hello', 'pacakge:hello/hello.dart')
+Identifier identifier(Scope scope, String name, [String importFrom]) {
+  return (scope ?? Scope.identity).identifier(name, importFrom);
+}
+
+/// Maintains an imported reference scope to avoid conflicts in generated code.
 ///
-/// __Example use__:
+/// ## Example
 ///     void useContext(Scope scope) {
-///       // Prints Identifier "i1.Foo"
-///       print(scope.getIdentifier('Foo', 'package:foo/foo.dart');
+///       // May print 'i1.Foo'.
+///       print(scope.identifier('Foo', 'package:foo/foo.dart'));
 ///
-///       // Prints Identifier "i1.Bar"
-///       print(scope.getIdentifier('Bar', 'package:foo/foo.dart');
+///       // May print 'i1.Bar'.
+///       print(scope.identifier('Bar'), 'package:foo/foo.dart');
```

Comment:
here and in the `Baz` example, the closing parenthesis should be after the package, not the identifier